src/Mod/Sketcher/Gui/TaskSketcherConstraints.cpp

```diff
@@ -1371,6 +1371,8 @@ void TaskSketcherConstraints::onSelectionChanged(const Gui::SelectionChanges& ms
                                 auto tmpBlock = ui->listWidgetConstraints->blockSignals(true);
                                 item->setSelected(select);
                                 ui->listWidgetConstraints->blockSignals(tmpBlock);
+                                QModelIndex index = ui->listWidgetConstraints->model()->index(i, 0);

```

Comment:
Can this "model()" ever give a `nullptr`? Does there need to be some safeguard against this?

Reply:
Sure we can check that it's not a nullptr, but if it is, maybe it's better to crash to expose the underlying bug? Don't know what the FreeCAD policy is on these things and on bloating the code with unnecessary checks (and on what is deemed "necessary"). What's the philosophy?

Reply:
There's no real policy in place for this. But if there where I think it would be something like: If something can be null or if you're unsure - add a check, especially if the codepath is an edge case.

Reply:
Looking more carefully, this seems to be entirely something handled by Qt. If anything `index(i, 0)` may have i out of bounds.

Reply:
That said, I'm OK with keeping the check since it's already added.

In general, I guess, as people have earlier suggested, if the `nullptr` is coming from FreeCAD, we should consider using `std::optional` and/or throwing an exception. For `nullptr` coming from dependencies, of course, we must use these checks.